app/src/main/java/com/example/parkingsystemkotlin/mvp/model/ReservationModel.kt

```diff
@@ -37,14 +38,44 @@ class ReservationModel(private val reservationInformationDB: DataBase) : Reserva
 
     override fun saveReservation(securityCode: String, place: String) {
         if (::startDateAndTime.isInitialized && ::startDateAndTime.isInitialized && securityCode.isNotEmpty() && place.isNotEmpty()) {
-            reservationInformationDB.putReservationDB(startDateAndTime, finishDateAndTime, securityCode, place)
+            if (!isOverlap(place)) {
+                reservationInformationDB.putReservationDB(startDateAndTime, finishDateAndTime, securityCode, place)
+            }
         }
     }
 
+    private fun isOverlap(place: String): Boolean {
+        reservationInformationDB.getReservationsDB(place).forEach {
+            if (!(startDateAndTime.after(it.finishDateAndTime) || finishDateAndTime.before(it.startDateAndTime))) {
+                overlap = true
+                return true
+            }
+        }
+        return false
+    }
+
```

Comment:
Instead to do this you can create a filter for the list that you are referencing inside DB.
You can return the result of the filter. The idea is if you find an element in which the condition you want to check is true, it returns true if it does not find any element, it would return false.

Reply:
done